py/desispec/fiberbitmasking.py

```diff
@@ -100,46 +105,48 @@ def get_fiberbitmasked_frame_arrays(frame,bitmask=None,ivar_framemask=True,retur
         return ivar
 
 
-def get_fiberbitmask_comparison_value(kind='fluxcalib'):
+def get_fiberbitmask_comparison_value(kind,band):
     """
         Takes a string argument and returns a 32-bit integer representing the logical OR of all
         relevant fibermask bits for that given reduction step
 
         input:
              kind: str : string designating which combination of bits to use based on the operation
-
+             band: str : ('b' 'r' or 'z')
         possible values are:
               "all", "sky" (or "skysub"), "flat", "flux" (or "fluxcalib"), "star" (or "stdstars")
     """
     if kind.lower() == 'all':
-        return get_all_fiberbitmask_val()
+        return get_all_fiberbitmask_with_amp(band)
     elif kind.lower()[:3] == 'sky':
-        return get_skysub_fiberbitmask_val()
+        return get_skysub_fiberbitmask_val(band)
     elif kind.lower() == 'flat':
-        return get_flat_fiberbitmask_val()
+        return get_flat_fiberbitmask_val(band)
     elif 'star' in kind.lower():
-        return get_stdstars_fiberbitmask_val()
+        return get_stdstars_fiberbitmask_val(band)
     elif 'flux' in kind.lower():
-        return get_fluxcalib_fiberbitmask_val()
+        return get_fluxcalib_fiberbitmask_val(band)
     else:
         log = get_logger()
         log.warning("Keyword {} given to get_fiberbitmask_comparison_value() is invalid.".format(kind)+\
                     " Using 'fluxcalib' fiberbitmask.")
-        return get_fluxcalib_fiberbitmask_val()
+        return get_fluxcalib_fiberbitmask_val(band)
 
 
-def get_skysub_fiberbitmask_val():
-    return get_all_fiberbitmask_val()
+def get_skysub_fiberbitmask_val(band):
+    return get_all_fiberbitmask_with_amp(band)
 
-def get_flat_fiberbitmask_val():
+def get_flat_fiberbitmask_val(band):
```

Comment:
It's unclear why we need to add a dummy parameter here, but it's okay for uniformity.

Reply:
I left it because in principle the flat_fiberbitmask should also be a function of the band.
It's just because we have decided to run the fiberflat even if BADAMP is set that this argument is not used for now.

Reply:
Thank you for the explanation